### PR TITLE
Add Whitelist/Blacklist mode for void miners

### LINKED ISSUE #9450: Implement a whitelist filter for Void miners
### Your GTNH Discord Username

Boldar#1676

### Your Pack Version

2.1.2.1

### Your Proposal

At later Tiers, void miner outputs are like 90% useless. Like digging for Infinity Catalyst on Vega B will generate 95% crap. It would be nice to have some sort of filter/card/whatever to add to the void miner to just not generate the trash ores, like a whitelist of what i actually want. This should not change the chances, if it would generate something not on the filter, just output nothing for that cycle.

This is how i solved it right now, 2 item filters, one inverted into thrash can and one to processing, but i think it would be better to just not generate the ore at all.

### Your Goal

Make Void miners more performant.

### Your Vision

It would help with getting just the ressources i want.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this feature please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I believe there is nothing similar in the pack already, or the existing solution isn't good enough.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members thinking it's inappropriate.
- [X] I believe this feature would make the pack better.

### PR DESCRIPTION
**Pull request by MuXiu1997**
Closed https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/9450

### CODE REVIEW

File: src/main/java/com/github/bartimaeusnek/crossmod/galacticgreg/GT_TileEntity_VoidMiner_Base.java [hunk 1]
```diff
@@ -415,8 +419,16 @@ private void makeDropMap() {
     }
 
     private void handleOutputs() {
-        Pair<Integer,Boolean> stats = getOreDamage();
-        this.addOutput(new ItemStack(stats.getValue() ? WerkstoffLoader.BWOres : GregTech_API.sBlockOres1, multiplier, stats.getKey()));
+        Pair<Integer, Boolean> stats = getOreDamage();
+        final ArrayList<ItemStack> inputs = this.getStoredInputs();
+        final ItemStack output = new ItemStack(stats.getValue() ? WerkstoffLoader.BWOres : GregTech_API.sBlockOres1, multiplier, stats.getKey());
+        if (inputs.size() == 0) {
```
Comment (GlodBlock) [inline src/main/java/com/github/bartimaeusnek/crossmod/galacticgreg/GT_TileEntity_VoidMiner_Base.java]: add a ore filter for this list, only ore block should be considered as whitelist

**Review by GlodBlock — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/bartimaeusnek/crossmod/galacticgreg/GT_TileEntity_VoidMiner_Base.java [hunk 2]
```diff
@@ -415,8 +419,16 @@ private void makeDropMap() {
     }
 
     private void handleOutputs() {
-        Pair<Integer,Boolean> stats = getOreDamage();
-        this.addOutput(new ItemStack(stats.getValue() ? WerkstoffLoader.BWOres : GregTech_API.sBlockOres1, multiplier, stats.getKey()));
+        Pair<Integer, Boolean> stats = getOreDamage();
+        final ArrayList<ItemStack> inputs = this.getStoredInputs();
```
Comment (GlodBlock) [inline src/main/java/com/github/bartimaeusnek/crossmod/galacticgreg/GT_TileEntity_VoidMiner_Base.java]: the void miner stilled won't output anything if someone left some mining pipes in the bus carelessly, you can change this into `final List<ItemStack> inputs = this.getStoredInputs().stream().filter(GT_Utility::isOre).collect(Collectors.toList());`

Comment (MuXiu1997) [reply src/main/java/com/github/bartimaeusnek/crossmod/galacticgreg/GT_TileEntity_VoidMiner_Base.java]: Careless

**Review by GlodBlock — APPROVED** [review]
looks good

**Comment by MuXiu1997** [comment]
@Dream-Master Made changes, need to re-review

**Pull request merged**